### PR TITLE
Support `ProjectConfig.dbt_project_path = None` & different paths for Rendering and Execution

### PR DESCRIPTION
**Pull request by MrBones757**
## Description

This MR finishes the work that was started in #605 to add full support for ProjectConfig.dbt_project_path = None, and implements #568.

Within this PR, several things have been updated:

1 - Added project_path fields to RenderConfig and ExecutionConfig
2 - Simplified the consumption of RenderConfig in the dbtGraph class
3 - added option to configure different dbt executable for Rendering vs Execution.

## Related Issue(s)

Closes: #568

## Breaking Change?

None

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by MrBones757** [comment]
Docs changes still need to be completed, though all the core logic and baseline tests in its current state has been implemented.

@tatiana could you please have a scan over the changes in this MR and let me know if there are any other changes you'd like to see before i finalize the docs / tests?

**Review by tatiana — COMMENTED** [review]
@MrBones757, this looks great, thank you!

I left two minor comments - I'll review them again after the checks are passed.

File: cosmos/config.py [hunk 1]
```diff
@@ -40,6 +40,13 @@ class RenderConfig:
     exclude: list[str] = field(default_factory=list)
     dbt_deps: bool = True
     node_converters: dict[DbtResourceType, Callable[..., Any]] | None = None
+    dbt_executable_path: str | Path = get_system_dbt()
+    dbt_project_path: InitVar[str | Path | None] = None
+
+    project_path: Path | None = field(init=False)
```
[the remark this answers is not in the document]
Comment (MrBones757) [reply cosmos/config.py:46]: I was experimenting with a feature of data classes here.
When the user creates an object here, they use dbt_project_path, but dbt_project_path is not an attribute of the class once constructed

dbt_project_path is only used for object generation and is passed directly to the __post_init__

project_path on the other hand, cant be defined using the init, but, it can be accessed as an object property, which, to your point will guarantee it being Path or None, simiar to what i did in ProjectConfig, without requiring a custom constructor.

**Comment by MrBones757** [comment]
I've added some additional tests to captures some more edge cases i found and add additional coverage.
Additionally, I've updated some docs, and docstrings, to reflect changes that have occurred in some Config Classes both within this PR and others.

All unit tests are running fine locally, and all integration tests other than databricks are too (no token for this)
I'll verify these in CI and once thats complete its looking just about complete.

**Comment by MrBones757** [comment]
Since the initial integration tests look good, is there anything else that needs to be done here :)
